src/kaleidoscope/device/dygma/raise/SideFlash.h

```diff
@@ -0,0 +1,85 @@
+/* -*- mode: c++ -*-
+ * kaleidoscope::device::dygma::Raise -- Kaleidoscope device plugin for Dygma Raise
+ * Copyright (C) 2017-2019  Keyboard.io, Inc
+ * Copyright (C) 2017-2019  Dygma Lab S.L.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#ifdef ARDUINO_SAMD_RAISE
+
+#include <Kaleidoscope.h>
+
+namespace kaleidoscope {
+namespace device {
+namespace dygma {
+namespace raise {
+
+template <typename _Firmware>
+class SideFlash : public kaleidoscope::Plugin {
+ private:
+  _Firmware firmware;
+ public:
+  EventHandlerResult onFocusEvent(const char *command) {
+    if (::Focus.handleHelp(command, PSTR("hardware.flash_left_side\nhardware.flash_right_side\nhardware.verify_left_side\nhardware.verify_right_side")))
+      return EventHandlerResult::OK;
+
+    if (strncmp_P(command, PSTR("hardware."), 9) != 0)
+      return EventHandlerResult::OK;
+
+    auto attiny = Kaleidoscope.device().attiny();
```

Comment:
What is this exactly?  the left and right side each have an attiny.

Reply:
Yes, but we use the same helper to flash them both, to save a bit of space. The only difference is the address. The naming is perhaps... not the best.

The idea is that the attiny _driver_ will help us flash or verify the firmware on either side, depending on the address.

Reply:
so, it's not an attiny flasher. it's a flasher for our custom i2c bootloader protocol. Better naming would help, I think.

Reply:
Aha. `kaleidoscope::util::flasher::keyboardio_i2c_bootloader`? A long name, but since it's only going to be used internally, it's ok.

Reply:
Yeah, that's fine.